test/datatype.test.js

```diff
@@ -52,6 +52,36 @@ describe('datatypes', function() {
     });
     Account.definition.properties.item.type.should.not.equal(String);
   });
+  it('should resolve array prop with connector specific metadata', function() {
+    const model = db.define('test', {
+      randomReview: {
+        type: [String],
+        mongodb: {
+          dataType: 'Decimal128',
+        },
+      },
+    });
+    model.definition.properties.randomReview.type.should.deepEqual(Array(String));
+    model.definition.properties.randomReview.mongodb.should.deepEqual({dataType: 'Decimal128'});
+  });
+
+  it('should coerce array of primitive types from string', async () => {
+    const dateArrayModel = db.define('dateArrayModel', {
+      bunchOfDates: [
+        {
+          type: Date,
+        },
+      ],
+    });
```

Comment:
On the second thought, I am not sure if this is a valid syntax. I remember we were recently fixing a problem where properties called `type` were treated incorrectly.

What are the rules for deciding whether this model contains an array of Date instances (`[dateVal, dateVal]`), vs. an array of anonymous objects that have a `type` property containing a date instance (`[{type: dateVal}, {type: dateVal}]`)?

Reply:
Good question. I think the fix we made recently allows users to use `type` name for properties IIRC. In my head, it'd be something like:
```js
prop: {
 type: [Date]
}
```
versus
```js
prop: {
 type: {
   type: [Date]
 }
}
```